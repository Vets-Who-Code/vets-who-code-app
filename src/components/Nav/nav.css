/* overide bootstrap mobile nav styles */

/* hamburger menu styles */
.hamburger {
  position: absolute;
  top: 15px;
  right: 8px;
}

.hamburger .line {
  width: 39px;
  height: 2px;
  background-color: #ecf0f1;
  display: block;
  margin: 8px auto;
  transition: all 0.3s ease-in-out;
}

.hamburger:hover {
  cursor: pointer;
}

.hamburger:focus {
  border: 1px dashed #fff;
}

#hamburger-1.is-active .line:nth-child(2) {
  opacity: 0;
}

#hamburger-1.is-active .line:nth-child(1) {
  transform: translateY(10px) rotate(45deg);
}

#hamburger-1.is-active .line:nth-child(3) {
  transform: translateY(-10px) rotate(-45deg);
}

.navbar-brand a {
  min-width: 200px;
  display: grid;
  grid-template-columns: 50px 180px;
  grid-template-rows: 50px;
}

.navbar-brand a .logo_holder {
  grid-column: 1/1;
  grid-row: 1/1;
}

.navbar-brand a .homeLink {
  grid-column: 2/2;
  grid-row: 1/1;
  padding-left: 5px;
  padding-top: 15px;
  vertical-align: middle;
  font-size: 0.5em;
}
