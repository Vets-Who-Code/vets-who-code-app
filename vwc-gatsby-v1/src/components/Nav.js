import React from 'react';
import Link from 'gatsby-link';
import logo from '../images/flag.gif';

const Nav = () => (
  <nav
    id="fixedTopNav"
    className="navbar navbar-fixed-top main-navigation"
    itemScope=""
    itemType="https://schema.org/SiteNavigationElement"
    role="navigation"
  >
    <div className="container">
      <div className="navbar-header">
        <button
          type="button"
          className="navbar-toggle collapsed"
          data-toggle="collapse"
          data-target="#main-nav-collapse"
          aria-expanded="false"
        >
          {' '}
          <span className="sr-only">#VetsWhoCode</span>{' '}
          <span className="ion-drag" />
        </button>
        <div
          className="navbar-brand"
          itemScope=""
          itemType="https://schema.org/Organization"
        >
          {' '}
          <span itemProp="name" className="sr-only">
            #VetsWhoCode
          </span>
          <a itemProp="url" href="index.html">
            <img src={logo} alt="#VetsWhoCode Logo" className="logo_holder" />
          </a>
        </div>
      </div>
      <div
        className="navbar-collapse collapse"
        id="main-nav-collapse"
        aria-expanded="false"
        style={{ height: '1px' }}
      >
        <ul className="nav navbar-nav navbar-right" role="menu">
          <li>
            {' '}
            <Link to="/">
              <span>Home</span>
            </Link>
          </li>
          <li>
            {' '}
            <Link to="/about">
              <span>About</span>
            </Link>
          </li>
          <li>
            {' '}
            <Link to="/syllabus">
              <span>Syllabus</span>
            </Link>
          </li>
          <li>
            {' '}
            <Link to="mentor">
              <span>Mentor</span>
            </Link>
          </li>
          <li>
            {' '}
            <Link to="apply">
              <span>Apply</span>
            </Link>
          </li>
          <li>
            {' '}
            <Link to="donate">
              <span>Donate</span>
            </Link>
          </li>
          <li>
            {' '}
            <Link to="contact">
              <span>Contact Us</span>
            </Link>
          </li>
          <li>
            {' '}
            <a href="https://medium.com/vets-who-code" target="_blank">
              <span>Blog</span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
);

export default Nav;
