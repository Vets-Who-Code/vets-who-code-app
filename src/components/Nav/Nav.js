import React, { useState, useRef, useEffect } from 'react'
import Link from 'gatsby-link'
import FluidImage from '../FluidImage'

import './nav.css'

function Nav() {
  const navRef = useRef()
  const [isNavOpen, setIsNavOpen] = useState(false)
  const [opacity, setOpacity] = useState(0.9)

  useEffect(() => {
    let current = true
    if (current) {
      document.addEventListener('scroll', handleScroll)
    }
    return () => {
      current = false
      document.removeEventListener('scroll', handleScroll)
    }
  })

  function handleScroll() {
    const winScroll = document.body.scrollTop || document.documentElement.scrollTop
    if (winScroll === 0) {
      setOpacity(0.9)
    }
    if (winScroll > 0) {
      setOpacity(1)
    }
  }

  return (
    <nav
      ref={navRef}
      id="fixedTopNav"
      className="navbar navbar-fixed-top main-navigation navbar-solid"
      itemScope=""
      itemType="https://schema.org/SiteNavigationElement"
      role="navigation"
      style={{ opacity: opacity }}
    >
      <div className="container">
        <div className="navbar-header">
          <div
            className="navbar-brand"
            style={{ padding: '5px 15px 5px' }}
            itemScope=""
            itemType="https://schema.org/Organization"
          >
            <span className="sr-only">#VetsWhoCode</span>
            <Link to="/">
              <FluidImage
                fileName="hashflag_white.jpg"
                alt="#VetsWhoCode Logo"
                className="logo_holder"
              />
              <div className="homeLink">VetsWhoCode</div>
            </Link>
          </div>
          <button
            type="button"
            id="hamburger-1"
            onClick={() => setIsNavOpen(!isNavOpen)}
            className={`navbar-toggle collapsed hamburger ${isNavOpen ? 'is-active' : ''}`}
            aria-expanded={isNavOpen ? 'true' : 'false'}
          >
            <span className="line"></span>
            <span className="line"></span>
            <span className="line"></span>
            <span className="sr-only">#VetsWhoCode</span>
          </button>
        </div>
        <div
          className={`navbar-collapse collapse ${isNavOpen ? 'in' : ''}`}
          id="main-nav-collapse"
          aria-expanded={isNavOpen ? 'true' : 'false'}
        >
          <ul className="nav navbar-nav navbar-right" role="menu">
            <li role="menuitem">
              <Link to="/">
                <span>Home</span>
              </Link>
            </li>
            <li role="menuitem">
              {' '}
              <Link to="/about">
                <span>About</span>
              </Link>
            </li>
            <li role="menuitem">
              {' '}
              <Link to="/syllabus">
                <span>Syllabus</span>
              </Link>
            </li>
            <li role="menuitem">
              <Link to="/mentor">
                <span>Mentor</span>
              </Link>
            </li>
            <li role="menuitem">
              <Link to="/apply">
                <span>Apply</span>
              </Link>
            </li>
            <li role="menuitem">
              <Link to="/donate">
                <span>Donate</span>
              </Link>
            </li>
            <li role="menuitem">
              <Link to="/contact">
                <span>Contact Us</span>
              </Link>
            </li>
            <li role="menuitem">
              <a
                href="https://medium.com/vets-who-code"
                without="true"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>Blog</span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  )
}

export default Nav
