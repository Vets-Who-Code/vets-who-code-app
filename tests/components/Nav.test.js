import { render } from '@testing-library/react'
import Nav from '@/components/Nav'

describe('<Nav />', () => {
  test('should render correctly', () => {
    const { container } = render(<Nav />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
