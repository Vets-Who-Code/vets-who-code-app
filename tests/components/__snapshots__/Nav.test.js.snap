// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render correctly 1`] = `
<nav
  class="navbar navbar-fixed-top main-navigation navbar-solid"
  id="fixedTopNav"
  itemtype="https://schema.org/SiteNavigationElement"
  role="navigation"
  style="opacity: 0.9;"
>
  <div
    class="container"
  >
    <div
      class="navbar-header"
    >
      <div
        class="navbar-brand"
        itemtype="https://schema.org/Organization"
        style="padding: 5px 15px 5px;"
      >
        <span
          class="sr-only"
        >
          #VetsWhoCode
        </span>
        <a
          aria-current="page"
          class=""
          href="/"
        >
          static-query
          <div
            class="homeLink"
          >
            VetsWhoCode
          </div>
        </a>
      </div>
      <button
        aria-expanded="false"
        class="navbar-toggle collapsed hamburger "
        id="hamburger-1"
        type="button"
      >
        <span
          class="line"
        />
        <span
          class="line"
        />
        <span
          class="line"
        />
        <span
          class="sr-only"
        >
          #VetsWhoCode
        </span>
      </button>
    </div>
    <div
      aria-expanded="false"
      class="navbar-collapse collapse "
      id="main-nav-collapse"
    >
      <ul
        class="nav navbar-nav navbar-right"
        role="menu"
      >
        <li
          role="menuitem"
        >
          <a
            aria-current="page"
            class=""
            href="/"
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li
          role="menuitem"
        >
           
          <a
            href="/about"
          >
            <span>
              About
            </span>
          </a>
        </li>
        <li
          role="menuitem"
        >
           
          <a
            href="/syllabus"
          >
            <span>
              Syllabus
            </span>
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            href="/mentor"
          >
            <span>
              Mentor
            </span>
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            href="/apply"
          >
            <span>
              Apply
            </span>
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            href="/donate"
          >
            <span>
              Donate
            </span>
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            href="/contact"
          >
            <span>
              Contact Us
            </span>
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            href="https://medium.com/vets-who-code"
            rel="noopener noreferrer"
            target="_blank"
            without="true"
          >
            <span>
              Blog
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
